src/libxtp/gwbse/sigma_cda.cc

```diff
@@ -0,0 +1,102 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <votca/tools/constants.h>
+#include <votca/xtp/gw.h>
+#include <votca/xtp/sigma_cda.h>
+
+namespace votca {
+namespace xtp {
+
+void Sigma_CDA::PrepareScreening() {
+  GaussianQuadrature::options opt;
+  opt.homo = _opt.homo;
+  opt.order = _opt.order;
+  opt.qptotal = _qptotal;
+  opt.qpmin = _opt.qpmin;
+  opt.rpamin = _opt.rpamin;
+  opt.quadrature_scheme = _opt.quadrature_scheme;
+  _gq.configure(opt, _rpa);
+}
+
+// This function is used in the calculation of the residues
+// This calculates eps^-1 (inverse of the dielectric function) for complex
+// frequencies of the kind omega = delta + i*eta
+double Sigma_CDA::CalcDiagContribution(const Eigen::RowVectorXd& Imx_row,
+                                       double delta, double eta) const {
+  std::complex<double> delta_eta(delta, eta);
+  Eigen::MatrixXd DielMxInv = _rpa.calculate_epsilon_r(delta_eta).inverse();
```

Comment:
actually, you never have to explicitly calculate the inverse directly. All you are interested in is how the inverse acts on a vector. So solve the linear system instead,  so basically you want A-1*b, which you can get from solving Ax=b

Reply:
Any suggestion about the preferred eigen method for this?

Reply:
I would use the cheapest first so `llt()` 

https://eigen.tuxfamily.org/dox/group__TutorialLinearAlgebra.html

Reply:
Is done in the residue contribution. Currently checking the performance of the different linear system solvers. According to https://eigen.tuxfamily.org/dox/group__DenseDecompositionBenchmark.html `PartialPivLU` should be the best choice in terms of speed.

Reply:
I tested this for big-ish molecule: QP corrections for 25 near-gap levels, 28 threads, 3555 aux basis functions, with fixed-point updating, so in both cases 149 self-energy evaluations.

Using `.inverse()`: 6 hours 53 minutes
Using `PartialPivLU`: 6 hours 38 minutes

Linear system is about 15 minutes faster, or around 3%.
Probably doesn't hurt to keep the linear system, but the gain is a little small.

Reply:
Yes, it is also much more stable numerically. So this was the main reason for me and it is a bit faster. If we can speed the RPA up, this will get faster too.

Reply:
OK, removed the old implementation here.